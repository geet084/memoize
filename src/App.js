import React, { Component } from 'react';
import './CSS/Main.scss';
import Display from './Display.js';
import Nav from './Nav.js';
import prototypes from './datasets/aType.js';

class App extends Component {
  constructor() {
    super();

    this.state = {
      prototypes: prototypes,
      selection: 'all'
    }
  }

  updateType = (e) => {
    if (e.target.value !== undefined) {
      this.setState({ selection: e.target.value })
    }
    if (e.target.value === 0) {
      this.setState({ selection: e.target.className })
    }
  }

  render() {
    let { prototypes, selection } = this.state;
    return (
      <main className="App">
        <h1>Welcome ....</h1>
        <Nav prototypes={prototypes}
             showCard={this.showCard}
             updateSelection={this.updateType} />
        {/* <Display prototypes={prototypes[selection]} /> */}
        <Display prototypes={prototypes}
                 selection={selection} />
      </main>
    );
  }
}

export default App;
