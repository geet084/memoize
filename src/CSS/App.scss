.App {
  // text-align: center;
  // background-color: lightblue;
  // height: 200px;
  // font-size: 2rem;
}