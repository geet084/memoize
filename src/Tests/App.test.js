import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { shallow } from 'enzyme';
global.localStorage = '../setupTests.js';

global.JSON = {
  parse: function (str) {
    return str;
  },
  stringify: function (str) {
    return str;
  }
}

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <App />
    )
    wrapper.setState({answeredQuestions: []})
  })

  afterEach(() => {
    localStorage.clear();
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should return an empty array if local storage is empty', () => { 
    
    expect(wrapper.instance().getFromStorage()).toEqual([])
  })

  it('should return and array of items from local storage if it is not empty', () => {

    localStorage.setItem('userAnsweredQuestions', ['a', 'b'])
    let newArray = localStorage.getItem('userAnsweredQuestions')
    expect(wrapper.instance().getFromStorage()).toEqual(newArray)    
  })

  it('should update answered questions list', () => {
    expect(wrapper.state('answeredQuestions')).toEqual([])
    expect(wrapper.instance().getFromStorage()).toEqual([])

    wrapper.instance().updateAnsweredQuestions({}, 'hello');

    expect(wrapper.state('answeredQuestions')).toEqual([{ userGuess: 'hello', question: {} }])
    let newArray = localStorage.getItem('userAnsweredQuestions')
    expect(wrapper.instance().getFromStorage()).toEqual(newArray)
  })

  it('should update the players score if the answer is guessed right', () => { 
    expect(wrapper.state('score')).toEqual(0)
    wrapper.setState({
      answeredQuestions: [{ userGuess: true, question: {} }]
    })
    wrapper.instance().tallyScore();
    expect(wrapper.state('score')).toEqual(1)
  })

  it('should not update the players score if the answer is guessed wrong', () => {
    expect(wrapper.state('score')).toEqual(0)
    wrapper.setState({
      answeredQuestions: [{ userGuess: false, question: {} }]
    })
    wrapper.instance().tallyScore();
    expect(wrapper.state('score')).toEqual(0)
  })

  it('should reset the score and answered questions list', () => {
    wrapper.setState({
      answeredQuestions: [{ score: 9, userGuess: false, question: {} }]
    })
    
    wrapper.instance().resetAnsweredQuestions();

    expect(wrapper.state()).toEqual({
      score: 0, answeredQuestions: [], prototypes: []
    })
  })
});

