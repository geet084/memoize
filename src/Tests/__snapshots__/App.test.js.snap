// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<main
  className="App"
>
  <h1
    className="header"
  >
    Welcome to Study Time
  </h1>
  <Nav
    resetHandler={[Function]}
    score={0}
  />
  <Display
    answeredQuestions={Array []}
    prototypes={Array []}
    updateAnswersHandler={[Function]}
  />
</main>
`;
